src/TLuaInterpreter.cpp

```diff
@@ -6801,30 +6844,19 @@ int TLuaInterpreter::exists( lua_State * L )
     QString name = _name.c_str();
     if( type == "timer")
     {
-        QMap<QString, TTimer *>::const_iterator it1 = pHost->getTimerUnit()->mLookupTable.find( name );
-        while( it1 != pHost->getTimerUnit()->mLookupTable.end() && it1.key() == name )
-        {
-            cnt++;
-            it1++;
-        }
+        cnt += pHost->getTimerUnit()->mLookupTable.count( name );
     }
     else if( type == "trigger")
     {
-        QMap<QString, TTrigger *>::const_iterator it1 = pHost->getTriggerUnit()->mLookupTable.find( name );
-        while( it1 != pHost->getTriggerUnit()->mLookupTable.end() && it1.key() == name )
-        {
-            cnt++;
-            it1++;
-        }
+        cnt += pHost->getTriggerUnit()->mLookupTable.count( name );
     }
     else if( type == "alias")
     {
-        QMap<QString, TAlias *>::const_iterator it1 = pHost->getAliasUnit()->mLookupTable.find( name );
-        while( it1 != pHost->getAliasUnit()->mLookupTable.end() && it1.key() == name )
-        {
-            cnt++;
-            it1++;
-        }
+        cnt += pHost->getAliasUnit()->mLookupTable.count( name );
+    }
+    else if( type == "keybind")
+    {
+        cnt += pHost->getKeyUnit()->mLookupTable.count( name );
```

Comment:
This is the right method to call - Mudlet by design allows you to have multiple items named the same, so an exists() that returns a boolean wouldn't be very useful because you'd also need to know how many if there are multiple. It could have been called count() or something like that - but since most of the time you're checking for existence of one item, it's called exists(), and it tells you _how many_ of an item exists, not _does_ it exist or not. So it still makes semantic sense in that regard when you take into account the context of multiple items being called the same.